src/register-evm-agent/src/register_evm_agent/Cargo.toml

```diff
@@ -31,6 +31,7 @@ hex = { workspace = true }
 ic-agent = { workspace = true }
 ic-exports = { workspace = true, features = ["state-machine"] }
 log = { workspace = true }
+openssl = {version="0.10.55", features = ["vendored"] } 
```

Comment:
why is this necessary?

Reply:
Openssl requires native libs. Thus there is a big possibility that linux build would work only for the platform only compatible with CI build OS. On different linux distros actual openssl libs could be located by different paths.

Reply:
We should verify what dependency requires openssl and check if there's a feature to use rustls instead

Reply:
Cargo tree shows openssl in dependencies. 

BTW this one reason why I just can't easily cross compile it. I need to cross compile  openssl for all target platforms too.